core/src/main/java/org/apache/calcite/runtime/ResultSetEnumerable.java

```diff
@@ -162,69 +165,81 @@ private ResultSetEnumerable(
     return new ResultSetEnumerable<>(dataSource, sql, rowBuilderFactory, consumer);
   }
 
-  /** Called from generated code that proposes to create a
-   * {@code ResultSetEnumerable} over a prepared statement. */
+  /**
+   * Called from generated code that proposes to create a
+   * {@code ResultSetEnumerable} over a prepared statement.
+   */
   public static PreparedStatementEnricher createEnricher(Integer[] indexes,
-      DataContext context) {
+                                                         DataContext context) {
     return preparedStatement -> {
       for (int i = 0; i < indexes.length; i++) {
         final int index = indexes[i];
         setDynamicParam(preparedStatement, i + 1,
-            context.get("?" + index));
+            (TypedValue) context.get("?" + index));
       }
     };
   }
 
-  /** Assigns a value to a dynamic parameter in a prepared statement, calling
-   * the appropriate {@code setXxx} method based on the type of the value. */
+  /**
+   * Assigns a value to a dynamic parameter in a prepared statement, calling
+   * the appropriate {@code setXxx} method based on the type of the value.
+   */
   private static void setDynamicParam(PreparedStatement preparedStatement,
-      int i, Object value) throws SQLException {
+                                      int i, TypedValue value) throws SQLException {
     if (value == null) {
```

Comment:
Is it the proper way to handle nulls?
I guess in case of nulls, `TypedValue value` will be non-null, and `null` will be stored inside.